crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -90,6 +111,58 @@ impl SyncTaskSupervisor {
         Self { task, termination_sender }
     }
 
+    async fn offline_check(
```

Comment:
Since this is not trivial, can you add a small doc comment explaining what it is?

Reply:
Sure: a685c463a67271b3654e24eb5648f3214926e598.